arrow-libs/core/arrow-autoclose/src/commonMain/kotlin/arrow/AutoCloseScope.kt

```diff
@@ -63,43 +63,43 @@ import kotlin.coroutines.cancellation.CancellationException
  */
 public inline fun <A> autoCloseScope(block: AutoCloseScope.() -> A): A {
   val scope = DefaultAutoCloseScope()
+  var finalized = false
   return try {
     block(scope)
-      .also { scope.close(null) }
-  } catch (e: CancellationException) {
```

Comment:
`throwIfFatal` is only used here, so I made it not consider CancellationExceptions fatal for the sake of brevity.

Reply:
Scratch that, I've gone back to explicitly singling out CancellationException